### PR TITLE
FLEX-3787 Remove metered functionality from the platform

### PR DESCRIPTION
**Pull request by bvdzwet**
Removes the metered boolean from DeviceModel, and removes the meter
type with short interval, long interval and long interval type from the
device configuration (setting and getting).

### CODE REVIEW

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/GetConfiguration.feature [hunk 1]
```diff
@@ -35,29 +31,25 @@ Feature: CoreConfigurationManagement GetConfiguration
...
+      | OSLP ELSTER | RELAY                   |          |         | 1,1,LIGHT  |                   | 10.20.30.40   |    12122 |
+      | OSLP ELSTER | ONE_TO_TEN_VOLT         |          |         |            |                   | 10.20.30.40   |    12122 |
+      | OSLP ELSTER | ONE_TO_TEN_VOLT_REVERSE |          |         |            |                   | 10.20.30.40   |    12123 |
+      | OSLP ELSTER | DALI                    |        2 | 1,2;2,1 |            |                   | 10.20.30.40   |    12123 |
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.40   |    12123 |
+      | OSLP ELSTER |                         |          |         |            | GPRS              | 10.20.30.50   |    12122 |
+      | OSLP ELSTER | DALI                    |          |         |            |                   | 10.20.30.50   |    12122 |
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   |    12122 |
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   |    12122 |
```
Comment (smvdheijden) [inline integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/GetConfiguration.feature:48]: Remove duplicate example.

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/GetConfiguration.feature [hunk 2]
```diff
@@ -35,29 +31,25 @@ Feature: CoreConfigurationManagement GetConfiguration
...
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   |    12123 |
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   |    12123 |
```
Comment (smvdheijden) [inline integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/GetConfiguration.feature:50]: Remove duplicate example.

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/SetConfiguration.feature [hunk 3]
```diff
@@ -32,38 +28,34 @@ Feature: CoreConfigurationManagement SetConfiguration
...
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.40   | 12123    |
+      | OSLP ELSTER |                         |          |         |            | GPRS              | 10.20.30.40   | 12123    |
+      | OSLP ELSTER | RELAY                   |          |         |            |                   | 10.20.30.40   | 12123    |
+      | OSLP ELSTER | DALI                    |          |         |            |                   | 10.20.30.40   | 12123    |
+      | OSLP ELSTER | RELAY                   |          |         | 1,1,LIGHT  |                   | 10.20.30.50   | 12122    |
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   | 12122    |
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   | 12122    |
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   | 12122    |
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   | 12122    |
```
Comment (smvdheijden) [inline integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/SetConfiguration.feature:53]: Remove these 3 duplicate examples.

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/SetConfiguration.feature [hunk 4]
```diff
@@ -32,38 +28,34 @@ Feature: CoreConfigurationManagement SetConfiguration
...
-      | OSLP ELSTER | DALI                    | 1        | 1,1     |            |                   |           |               |              |              | 10.20.30.50   | 12123    |
-      | OSLP ELSTER | DALI                    | 2        | 1,1;2,2 |            |                   |           |               |              |              | 10.20.30.50   | 12123    |
-      | OSLP ELSTER |                         |          |         |            |                   |           |               | 10           |              | 10.20.30.50   | 12123    |
+      | Protocol    | LightType               | DcLights | DcMap   | RelayConf  | PreferredLinkType | OsgpIpAddress | OsgpPort |
+      | OSLP ELSTER | RELAY                   |          |         |            |                   | 10.20.30.40   | 12122    |
+      | OSLP ELSTER | RELAY                   |          |         | 1,1,TARIFF |                   | 10.20.30.40   | 12122    |
+      | OSLP ELSTER | RELAY                   |          |         | 1,1,TARIFF |                   | 10.20.30.40   | 12122    |
```
Comment (smvdheijden) [inline integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/SetConfiguration.feature:41]: Remove duplicate example.

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/SetConfiguration.feature [hunk 5]
```diff
@@ -32,38 +28,34 @@ Feature: CoreConfigurationManagement SetConfiguration
...
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   | 12122    |
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   | 12122    |
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   | 12122    |
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   | 12122    |
+      | OSLP ELSTER | RELAY                   |          |         | 1,1,LIGHT  | CDMA              | 10.20.30.50   | 12123    |
+      | OSLP ELSTER | RELAY                   |          |         | 1,1,LIGHT  | ETHERNET          | 10.20.30.50   | 12123    |
+      | OSLP ELSTER | DALI                    | 1        | 1,1     |            |                   | 10.20.30.50   | 12123    |
+      | OSLP ELSTER | DALI                    | 2        | 1,1;2,2 |            |                   | 10.20.30.50   | 12123    |
+      | OSLP ELSTER |                         |          |         |            |                   | 10.20.30.50   | 12123    |
```
Comment (smvdheijden) [inline integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/SetConfiguration.feature:58]: Remove duplicate example.

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/entities/DeviceModel.java [hunk 6]
```diff
@@ -37,28 +37,22 @@
...
     }
 
-    public DeviceModel(final Manufacturer manufacturer, final String modelCode, final String description,
-            final boolean metered) {
+    public DeviceModel(final Manufacturer manufacturer, final String modelCode, final String description) {
         this.manufacturer = manufacturer;
         this.modelCode = modelCode;
         this.description = description;
         // default behavior is true
         this.fileStorage = true;
```
Comment (smvdheijden) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/entities/DeviceModel.java:49]:
```suggestion
        this(manufacturer, modelCode, description, true);
```

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/entities/DeviceModel.java [hunk 7]
```diff
@@ -37,28 +37,22 @@
...
-        this.metered = metered;
     }
 
     public DeviceModel(final Manufacturer manufacturer, final String modelCode, final String description,
-            final boolean fileStorage, final boolean metered) {
-        this(manufacturer, modelCode, description, metered);
+            final boolean fileStorage) {
+        this(manufacturer, modelCode, description);
```
Comment (smvdheijden) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/entities/DeviceModel.java:54]:
```suggestion
        this.manufacturer = manufacturer;
        this.modelCode = modelCode;
        this.description = description;
```

**Review by smvdheijden — CHANGES_REQUESTED** [review]
"Grepping around" in this branch revealed several files that possibly still might need some attention?

$ grep -ir metered

integration-tests/cucumber-tests-platform-common/soap-ui-project/Core-SoapUI-project.xml

$ grep -ir historyinterval

integration-tests/cucumber-tests-platform-common/soap-ui-project/Core-SoapUI-project.xml

$ grep -ir metertype

integration-tests/cucumber-tests-platform-publiclighting/soap-ui-project/PublicLighting-SoapUI-project.xml
integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-publiclighting/DeviceMonitoring/GetPowerUsageHistory.feature
integration-tests/cucumber-tests-platform-publiclighting/src/test/java/org/opensmartgridplatform/cucumber/platform/publiclighting/PlatformPubliclightingDefaults.java
integration-tests/cucumber-tests-platform-publiclighting/src/test/java/org/opensmartgridplatform/cucumber/platform/publiclighting/mocks/oslpdevice/MockOslpServer.java
integration-tests/cucumber-tests-platform-publiclighting/src/test/java/org/opensmartgridplatform/cucumber/platform/publiclighting/glue/steps/ws/publiclighting/devicemonitoring/GetPowerUsageHistorySteps.java
integration-tests/cucumber-tests-platform-common/soap-ui-project/Core-SoapUI-project.xml
integration-tests/cucumber-tests-platform/src/test/java/org/opensmartgridplatform/cucumber/platform/PlatformKeys.java
osgp/platform/osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/mapping/PowerUsageDataConverter.java
osgp/platform/osgp-adapter-ws-publiclighting/src/main/java/org/opensmartgridplatform/adapter/ws/publiclighting/application/mapping/DeviceMonitoringMapper.java
osgp/platform/osgp-adapter-ws-publiclighting/src/main/java/org/opensmartgridplatform/adapter/ws/publiclighting/application/mapping/ws/PowerUsageMeterTypeConverter.java
osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/MeterType.java
osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/PowerUsageData.java
osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/mapping/DomainCoreMapper.java
osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/mapping/MeterTypeConverter.java
osgp/shared/osgp-dto/src/main/java/org/opensmartgridplatform/dto/valueobjects/PowerUsageDataDto.java
osgp/shared/osgp-dto/src/main/java/org/opensmartgridplatform/dto/valueobjects/MeterTypeDto.java
osgp/shared/osgp-ws-publiclighting/src/main/resources/schemas/pl-devicemonitoring.xsd
osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/infra/networking/services/commands/Iec61850PowerUsageHistoryCommand.java
osgp/protocol-adapter-oslp/osgp-adapter-protocol-oslp-elster/src/main/java/org/opensmartgridplatform/adapter/protocol/oslp/elster/application/mapping/OslpMapper.java
osgp/protocol-adapter-oslp/osgp-adapter-protocol-oslp-elster/src/main/java/org/opensmartgridplatform/adapter/protocol/oslp/elster/application/mapping/MeterTypeConverter.java
osgp/protocol-adapter-oslp/osgp-adapter-protocol-oslp-elster/src/main/java/org/opensmartgridplatform/adapter/protocol/oslp/elster/application/mapping/PowerUsageDataConverter.java
osgp/protocol-adapter-oslp/web-device-simulator/src/main/java/org/opensmartgridplatform/webdevicesimulator/service/OslpChannelHandler.java

**Comment by smvdheijden** [comment]
I believe GetPowerUsageHistory web service could be removed (along with some remnants of GetActualPowerUsage) since it is related to a devicemodel "being metered" and was only used for the removed web pages? This probably means that the public lighting device monitoring wsdl/xsd could be entirely removed...

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/entities/DeviceModel.java (hunk 6 above)
Comment (kevinsmeets) [reply osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/entities/DeviceModel.java:49]: It is not possible to have constructor a call constructor b, while constructor b calls constructor a. Not changing this code.

Comment (kevinsmeets) [reply osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/entities/DeviceModel.java:49]: Updated.

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/entities/DeviceModel.java (hunk 7 above)
Comment (kevinsmeets) [reply osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/entities/DeviceModel.java:54]: Updated.

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/SetConfiguration.feature (hunk 4 above)
Comment (kevinsmeets) [reply integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/SetConfiguration.feature:41]: All examples are unique now.

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/SetConfiguration.feature (hunk 5 above)
Comment (kevinsmeets) [reply integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/SetConfiguration.feature:58]: All examples are unique now.

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/SetConfiguration.feature (hunk 3 above)
Comment (kevinsmeets) [reply integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/SetConfiguration.feature:53]: All examples are unique now.

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/GetConfiguration.feature (hunk 2 above)
Comment (kevinsmeets) [reply integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/GetConfiguration.feature:50]: All examples are unique now.

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/GetConfiguration.feature (hunk 1 above)
Comment (kevinsmeets) [reply integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/ConfigurationManagement/GetConfiguration.feature:48]: All examples are unique now.

**Review by smvdheijden — APPROVED** [review]

**Review by smvdheijden — APPROVED** [review]

**Pull request merged**